azure/services/subnets/subnets.go

```diff
@@ -18,150 +18,111 @@ package subnets
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
 	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/pkg/errors"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
+const serviceName = "subnets"
+
 // SubnetScope defines the scope interface for a subnet service.
 type SubnetScope interface {
 	azure.ClusterScoper
-	SubnetSpecs() []azure.SubnetSpec
+	azure.AsyncStatusUpdater
+	UpdateSubnetID(string, string)
+	UpdateSubnetCIDRs(string, []string)
+	SubnetSpecs() []azure.ResourceSpecGetter
 }
 
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope SubnetScope
 	Client
+	async.Reconciler
 }
 
 // New creates a new service.
 func New(scope SubnetScope) *Service {
+	Client := NewClient(scope)
 	return &Service{
-		Scope:  scope,
-		Client: NewClient(scope),
+		Scope:      scope,
+		Client:     Client,
```

Comment:
is Client still needed?

Reply:
Yes, it's needed for the bastion host service, but the bastion host PR will remove the client when it merges.

Reply:
We need it for now, but the bastion host PR should remove it since that change removes the call.

Reply:
we need the subnet Client interface but do we need it in the subnet Service struct? I don't see it being used anywhere in subnets.go